### PR TITLE
Use Visual Studio colors and styles for all dialogs

### LINKED ISSUE #1785: PR filter text too dark on dark theme
__Version__
GitHub Extension for Visual Studio version: [2.5.4.3295](https://ci.appveyor.com/project/github-windows/visualstudio/build/2.5.4.3295/artifacts)

__What happened__
Using the dark theme, select a user from the assignable users filter. Open up the filter again and notice the unreadable `Clear filter` text and `X`
clear filter text too dark

### LINKED ISSUE #2300: GitHub dialogs use common VS Styles and themes
GitHub dialogs currently use a unique set of styles, colors, fonts and themes from the rest of Visual Studio. Suggestion is to move these dialogs to use VS' color service, themes and other elements to delver an experience that is visually integrated with the IDE bu maintains GitHub's brand.

image

### PR DESCRIPTION
**Pull request by jcansdale**
Fixes #2300

Migrate from using styles in `GitHub.UI` (which were inherited from GitHub Desktop) to using a combination of native Visual Studio styles and styles in `GitHub.VisualStudio.UI`. The styles in `GitHub.VisualStudio.UI` contain skins for custom GitHub controls and know about the Blue, Dark and Light Visual Studio themes.

### What this PR does

The following dialogs have been updated to use Visual Studio / `GitHub.VisualStudio.UI` styles:

- Login credentials
- Login 2FA
- Open from GitHub
- Create a repository
- Create a Gist
- Log out required
- Fork a repository

Allows previewing of views in designer with Visual Studio styles:

image

Other cleanup:
- Removed unused styles/controls from `GitHub.UI`

### To do

- [x] Fix when running with `Dark` theme
- [x] Fix info icon color
- [x] Fix the separator color (e.g. under `Sign in`)
- [x] Fix the progress bar colors
image
- [x] Fix the user filter box
image
Fixes #1785
Done:
image
- [x] Fix close icon of alert dialog
image
Here's how it should look:
image
Fixed:
image
image
- [x] Fix the PR list item background (it should be solid not an outline)

### Reference

- https://blog.slaks.net/2014-09-04/getting-started-with-visual-studio-themes/
- https://vsuiguide.azurewebsites.net/
- Scroll bars
https://vsuiguide.azurewebsites.net/topics/03%20Colors%20and%20Styling/03.03.05%20Using%20themed%20controls.html

### Reviewers

I've included screen grabs of the updated dialogs below. @donokuda are there any layout tweaks you think are now necessary?

A spinning animation no longer appears when a comment is in progress (e.g. when logging in). The button does however become inactive when the comment is running and this is hopefully indication enough that something is happening. We should check this works as expected when logging in and when creating a repository.

We should check this is working for dark and light themes.

We should check this is working on all supported versions of Visual Studio.

### Before and after

Before:
image

After:
image

#### Light theme

image

#### Dark theme

image

### Dark theme

Login credentials:
image

Login 2FA:
image

Open from GitHub:
image

Create a repository:

image

Create a Gist:
image

Fork a repository:
image

image

Leaving an inline comment

image

### Updated dialogs

Login credentials:
image

Login 2FA:
image

Create a repository:
image

Create a Gist:
image

Log out required - how do I make this dialog appear?

### CODE REVIEW

**Comment by jcansdale** [comment]
Looks like I have some themeing to fix:

File: src/GitHub.VisualStudio.UI/Views/Dialog/RepositoryCreationView.xaml [hunk 1]
```diff
@@ -60,19 +62,18 @@
             FocusVisualStyle="{x:Null}">
 
         <Grid.Resources>
-          <Style TargetType="{x:Type TextBlock}">
+                    <Style TargetType="{x:Type TextBlock}" BasedOn="{StaticResource {x:Static ui:VsResourceKeys.ThemedDialogTextBoxStyleKey}}">
```
Comment (grokys) [inline src/GitHub.VisualStudio.UI/Views/Dialog/RepositoryCreationView.xaml]: The target type here is for `TextBlock` and you seem to be applying a `TextBox` style to it.

**Comment by jcansdale** [comment]
Issue with spans and selected text?

image

**Comment by jcansdale** [comment]
Fix the list item background. It should look like this (circled).
image

**Pull request merged**